homeassistant/components/sensor/nederlandse_spoorwegen.py

```diff
@@ -0,0 +1,136 @@
+        vol.Required(CONF_NAME): cv.string,
+        vol.Required(CONF_FROM): cv.string,
+        vol.Required(CONF_TO): cv.string,
+        vol.Optional(CONF_VIA): cv.string}]
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the departure sensor."""
+    import ns_api
+    nsapi = ns_api.NSAPI(
+        config.get(CONF_EMAIL), config.get(CONF_PASSWORD))
+    sensors = []
+    for departure in config.get(CONF_DEPARTURES):
+        sensors.append(
+            NSDepartureSensor(
+                nsapi, departure.get(CONF_NAME), departure.get(CONF_FROM),
+                departure.get(CONF_TO), departure.get(CONF_VIA)))
+    add_devices(sensors)
+
+
+class NSDepartureSensor(Entity):
+    """Implementation of a NS Departure Sensor."""
+
+    def __init__(self, nsapi, name, departure, heading, via):
+        """Initialize the sensor."""
+        self._nsapi = nsapi
+        self._name = name
+        self._departure = departure
+        self._via = via
+        self._heading = heading
+        self._state = None
+        self._trips = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon for the frontend."""
+        return ICON
+
+    @property
+    def state(self):
+        """Return the next departure time."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        if not self._trips:
+            return
+
+        if self._trips[0].trip_parts:
+            route = [self._trips[0].departure]
+            for k in self._trips[0].trip_parts:
+                route.append(k.destination)
+
+        return {
+            'going': self._trips[0].going,
+            'departure_time_planned':
+                self._trips[0].departure_time_planned.strftime('%H:%M'),
+            'departure_time_actual':
+                self._trips[0].departure_time_actual.strftime('%H:%M'),
+            'departure_delay':
+                self._trips[0].departure_time_planned !=
+                self._trips[0].departure_time_actual,
+            'arrival_time_planned':
+                self._trips[0].arrival_time_planned.strftime('%H:%M'),
+            'arrival_time_actual':
+                self._trips[0].arrival_time_actual.strftime('%H:%M'),
+            'arrival_delay':
+                self._trips[0].arrival_time_planned !=
+                self._trips[0].arrival_time_actual,
+            'next':
+                self._trips[1].departure_time_actual.strftime('%H:%M'),
+            'status': self._trips[0].status.lower(),
+            'transfers': self._trips[0].nr_transfers,
+            'route': route,
+            'remarks': [r.message for r in self._trips[0].trip_remarks],
+            ATTR_ATTRIBUTION: CONF_ATTRIBUTION,
+        }
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
```

Comment:
According to your configuration you allow multiple entries. I don't know how the `nsapi` module works but usually it's smarter to use a data object to avoid multiple requests.

Reply:
Not entirely sure how to handle this. The API queries a single route (from - (via?) - to) and returns data. You can't lookup multiple rounds in one call, so I guess I'll leave this as is for now.